File: lib/features/club_setup/utils/club_objectives_ui.dart
import 'package:flutter/material.dart';
import 'package:viro_team_v2/config/viro_icons.dart';
import 'package:viro_team_v2/constants/firestore_fields.dart';

/// Icônes UI pour les objectifs club (assistant création).
IconData clubObjectiveIcon(String key) => switch (key) {
      ClubObjectives.planning => ViroIcons.calendar,
      ClubObjectives.attendance => ViroIcons.check,
      ClubObjectives.fees => ViroIcons.payments,
      ClubObjectives.equipment => ViroIcons.ball,
      ClubObjectives.communication => ViroIcons.bell,
      ClubObjectives.members => ViroIcons.users,
      ClubObjectives.teams => ViroIcons.groups,
      ClubObjectives.documents => ViroIcons.note,
      ClubObjectives.parents => ViroIcons.roleParent,
      ClubObjectives.stats => ViroIcons.trophy,
      _ => ViroIcons.check,
    };

File: portal/src/lib/clubSetup/clubObjectivesUi.ts
import { ClubObjectives } from "./constants";

/** Symbole compact par objectif club (équivalent icônes Phosphor mobile). */
export function clubObjectiveSymbol(key: string): string {
  switch (key) {
    case ClubObjectives.planning:
      return "📅";
    case ClubObjectives.attendance:
      return "✓";
    case ClubObjectives.fees:
      return "€";
    case ClubObjectives.equipment:
      return "⚽";
    case ClubObjectives.communication:
      return "🔔";
    case ClubObjectives.members:
      return "👥";
    case ClubObjectives.teams:
      return "🏟";
    case ClubObjectives.documents:
      return "📄";
    case ClubObjectives.parents:
      return "👨‍👩‍👧";
    case ClubObjectives.stats:
      return "🏆";
    default:
      return "✓";
  }
}
